src/test/protocol/KnownFormatToGRPC_test.cpp

```diff
@@ -0,0 +1,924 @@
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/TxFormats.h>
+
+#include "org/xrpl/rpc/v1/ledger_objects.pb.h"
+#include "org/xrpl/rpc/v1/transaction.pb.h"
+
+#include <cctype>
+#include <map>
+#include <string>
+#include <type_traits>
+
+namespace ripple {
+
+// This test suite uses the google::protobuf::Descriptor class to do runtime
+// reflection on our gRPC stuff.  At the time of this writing documentation
+// for Descriptor could be found here:
+//
+// https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.descriptor#Descriptor
+
+class KnownFormatToGRPC_test : public beast::unit_test::suite
+{
+private:
+    static constexpr auto fieldTYPE_UINT32 =
+        google::protobuf::FieldDescriptor::Type::TYPE_UINT32;
+
+    static constexpr auto fieldTYPE_UINT64 =
+        google::protobuf::FieldDescriptor::Type::TYPE_UINT64;
+
+    static constexpr auto fieldTYPE_BYTES =
+        google::protobuf::FieldDescriptor::Type::TYPE_BYTES;
+
+    static constexpr auto fieldTYPE_STRING =
+        google::protobuf::FieldDescriptor::Type::TYPE_STRING;
+
+    static constexpr auto fieldTYPE_MESSAGE =
+        google::protobuf::FieldDescriptor::Type::TYPE_MESSAGE;
+
+    // Format names are CamelCase and FieldDescriptor names are snake_case.
+    // Convert from CamelCase to snake_case.  Do not be fooled by consecutive
+    // capital letters like in NegativeUNL.
+    static std::string
+    formatNameToEntryTypeName(std::string fmtName)
+    {
+        std::string entryName;
+        entryName.reserve(fmtName.size());
+        bool prevUpper = false;
+        for (std::size_t i = 0; i < fmtName.size(); i++)
+        {
+            char const ch = fmtName[i];
+            bool const upper = std::isupper(ch);
+            if (i > 0 && !prevUpper && upper)
+                entryName.push_back('_');
+
+            prevUpper = upper;
+            entryName.push_back(std::tolower(ch));
+        }
+        return entryName;
+    };
+
+    // Create a map of (most) all the SFields in an SOTemplate.  This map
+    // can be used to correlate a gRPC Descriptor to its corresponding SField.
+    template <typename KeyType = int>
+    static std::map<std::string, SField const*>
+    soTemplateToSFields(
+        SOTemplate const& soTemplate,
+        [[maybe_unused]] KeyType fmtId)
+    {
+        std::map<std::string, SField const*> sFields;
+        for (SOElement const& element : soTemplate)
+        {
+            SField const& sField = element.sField();
+
+            // Fields that gRPC never includes.  This is probably intentional.
+            using FieldCode_t =
+                std::remove_const<decltype(SField::fieldCode)>::type;
+            static const std::set<FieldCode_t> excludedSFields{
+                sfLedgerIndex.fieldCode,
+                sfLedgerEntryType.fieldCode,
+                sfOperationLimit.fieldCode};
+
+            if (excludedSFields.count(sField.fieldCode) == 1)
```

Comment:
There are lots of places that follow this pattern. Either `if (someSetOrMap(). count(key) == 1)` or ` != 1`. Instead of this, I'd treat the return value of `count` as a bool and leave off the `== 1` or `!= 1`. Rational: we're using `count` as a stand-in for `contains`. We don't care about the actual count, just if the collection contains the key or not.